packages/core/src/repositories/repository.ts

```diff
@@ -59,6 +59,29 @@ export default abstract class Repository<
     return Result.ok(result[0]! as T)
   }
 
+  async findMany(ids: (string | number)[]) {
+    const result = await this.db
+      .select()
+      .from(this.scope)
+      // @ts-expect-error
+      .where(inArray(this.scope.id, ids))
+      .limit(ids.length)
+
+    if (!result.length) {
+      const missingIds = ids
+        // @ts-expect-error
+        .filter((id) => !result.find((r) => r.id === id))
+        .map((id) => "'" + id + "'")
+        .join(', ')
+
+      return Result.error(
+        new NotFoundError(`Records with ids ${missingIds} not found`),
+      )
+    }
```

Comment:
hmm a bit surprised this is in base repository since it has some funky behaviour i dont expect a findmany to fail if it does not find all of the ids provided

Reply:
hmm, really? I'd expect it to return an array of N items when I entered an array of N ids

Reply:
Easy change though